apps/prairielearn/src/pages/instructorAssessmentStatistics/instructorAssessmentStatistics.ts

```diff
@@ -1,26 +1,35 @@
-// @ts-check
 import * as express from 'express';
-const asyncHandler = require('express-async-handler');
+import asyncHandler = require('express-async-handler');
 import { stringify } from '@prairielearn/csv';
 
 import * as error from '@prairielearn/error';
 import { assessmentFilenamePrefix } from '../../lib/sanitize-name';
 import * as sqldb from '@prairielearn/postgres';
 import { updateAssessmentStatistics } from '../../lib/assessment';
+import { AssessmentSchema } from '../../lib/db-types';
+import {
+  AssessmentScoreHistogramByDateSchema,
+  DurationStatSchema,
+  InstructorAssessmentStatistics,
+  UserScoreSchema,
+} from './instructorAssessmentStatistics.html';
 
 const router = express.Router();
 const sql = sqldb.loadSqlEquiv(__filename);
 
+let scoreStatsCsvFilename = '';
+let durationStatsCsvFilename = '';
+let statsByDateCsvFilename = '';
 const setFilenames = function (locals) {
   const prefix = assessmentFilenamePrefix(
     locals.assessment,
     locals.assessment_set,
     locals.course_instance,
     locals.course,
   );
-  locals.scoreStatsCsvFilename = prefix + 'score_stats.csv';
-  locals.durationStatsCsvFilename = prefix + 'duration_stats.csv';
-  locals.statsByDateCsvFilename = prefix + 'scores_by_date.csv';
+  scoreStatsCsvFilename = prefix + 'score_stats.csv';
+  durationStatsCsvFilename = prefix + 'duration_stats.csv';
+  statsByDateCsvFilename = prefix + 'scores_by_date.csv';
 };
```

Comment:
These should not be global variables, otherwise two different users might have race conditions to access these values. One option is to return these values as an object instead.

Reply:
This is a dealbreaker, I'll block on this. Returning an object is exactly the right thing to do, see an example here:

https://github.com/PrairieLearn/PrairieLearn/blob/901fce8f19136e3c58fcdd45e9a0524191b475d4/apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.ts#L18

Reply:
Updated this to return an object in [f903993](https://github.com/PrairieLearn/PrairieLearn/pull/9809/commits/f903993f70436b07f9d110f5ad33ef7a39acccf7).